client.go

```diff
@@ -81,35 +83,48 @@ func importAccount(walletPath, secret string) (*wallet.Wallet, *wallet.Account,
 		}
 	}
 
-	w, err := wallet.NewWallet(ks, "")
+	w, err := ethwallet.NewWallet(ks, "")
 	if err != nil {
 		return nil, nil, errors.WithMessage(err, "creating wallet")
 	}
 
-	wAcc := wallet.NewAccountFromEth(w, &ethAcc)
+	wAcc := ethwallet.NewAccountFromEth(w, &ethAcc)
 	acc, err := w.Unlock(wAcc.Address())
-	return w, acc.(*wallet.Account), err
+	return w, acc.(*ethwallet.Account), err
 }
 
-func (c *Client) AddPeer(perunID *Address, ip string, port int) {
-	c.dialer.Register((*wallet.Address)(&perunID.Address), fmt.Sprintf("%s:%d", ip, port))
+func (c *Client) AddPeer(perunID *Address, host string, port int) {
+	c.dialer.Register((*ethwallet.Address)(&perunID.addr), fmt.Sprintf("%s:%d", host, port))
 }
 
-func (c *Client) ProposeChannel(ctx *Context, perunID *Address, initialBals *Allocation, challengeDuration int64) (*PaymentChannel, error) {
+func (c *Client) ProposeChannel(ctx *Context, perunID *Address, challengeDuration int64, initialBals *BigInts) (*PaymentChannel, error) {
+	if initialBals.Length() != 2 {
+		return nil, errors.New("initBals needs exactly two entries")
+	}
 	offAcc := c.w.NewAccount()
```

Comment:
I'd move `offAcc` and `initData` directly in the `prop` definition, since they're not used anywhere else...